web/src/routes/RepositoryDetails/Tags/TagsActionsAddTagModal.tsx

```diff
@@ -1,60 +1,127 @@
 import {
+  Alert,
   Button,
   Modal,
   ModalVariant,
+  Spinner,
   TextInput,
   Title,
 } from '@patternfly/react-core';
-import {useEffect, useState} from 'react';
+import {useCallback, useEffect, useState} from 'react';
 import {AlertVariant} from 'src/atoms/AlertState';
 import {useAlerts} from 'src/hooks/UseAlerts';
 import {useCreateTag} from 'src/hooks/UseTags';
 import {isNullOrUndefined} from 'src/libs/utils';
+import {getTags} from 'src/resources/TagResource';
+
+// Simple debounce function
+function debounce<T extends (...args: any[]) => void>(
+  func: T,
+  wait: number,
+): T {
+  let timeout: NodeJS.Timeout;
+  return ((...args: Parameters<T>) => {
+    clearTimeout(timeout);
+    timeout = setTimeout(() => func(...args), wait);
+  }) as T;
+}
 
 export default function AddTagModal(props: AddTagModalProps) {
   const [value, setValue] = useState('');
+  const [existingTagInfo, setExistingTagInfo] = useState<{
+    exists: boolean;
+    manifest?: string;
+  } | null>(null);
+  const [isCheckingTag, setIsCheckingTag] = useState(false);
   const {addAlert} = useAlerts();
   const {createTag, successCreateTag, errorCreateTag} = useCreateTag(
     props.org,
     props.repo,
   );
 
+  // Function to check if tag exists
+  const checkTagExists = async (tagName: string) => {
+    if (!tagName.trim()) {
+      setExistingTagInfo(null);
+      return;
+    }
+
+    setIsCheckingTag(true);
+    try {
+      // Call existing tags API to check if tag exists
+      const tagsResponse = await getTags(
```

Comment:
My only concern here is the list tags operation is quite expensive. I don't think it would cause any issues on quay.io since this isn't ran very often but it's something we probably shouldn't add if we don't need to. Does the old UI perform this same operation?

Reply:
Just tested on the old ui, it does not make these requests